### DISCUSSION TITLE
Performance of PnP core SDK vs CSOM Framework

### DISCUSSION THREAD

**Discussion by manjunathreddy1** [pnp/pnpcore#1153] [General]
Hi All,

When I was comparing the performance of PnP core SDK and CSOM Framework I noticed that to load the web object as mentioned in the below screenshot it is taking around 331 milliseconds using CSOM and it is taking around 2051 milliseconds using PnP core. Since it is recommended to use to PnP core for new projects I'm confused after seeing the time taken to load SharePoint object. 

Can anyone please confirm if this loading time expected or I'm doing some thing wrong. I was under the impression that loading SharePoint objects will be be faster in PnP core SDK since it is using Graph APIs.

![image](https://user-images.githubusercontent.com/265137/229289861-b14c4aac-67db-4023-bd70-47e9c3e7a4b8.png)

Manjunath

**Comment by jansenbe** [comment]
@manjunathreddy1 : that's weird. PnP Core uses a lot of REST, some Graph and little bit of CSOM under the covers. This is simply because Graph is not having the needed features (yet). Can you play around with the option to turn Graph on/off in the configuration and see if that makes a difference (https://pnp.github.io/pnpcore/using-the-sdk/basics-apis.html and https://pnp.github.io/pnpcore/using-the-sdk/basics-settings.html)? Also there's a fundamental difference between CSOM/Core SDK: when a context is created we already do two calls as we're caching some data
